src/Stateless/MethodDescription.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+
+namespace Stateless
+{
+    /// <summary>
+    /// Describes a method - either an action (activate, deactivate, etc.) or a transition guard
+    /// </summary>
+    public class MethodDescription
```

Comment:
MethodDescription is what the Stateless classes (ActivateActionBehaviour, EntryActionBehaviour, etc.) use internally to keep track of information about the actions; even though it's only exposed in the reflection API it's used all over the place in the base namespace so I don't think it makes sense to move it to Reflection.